lib/core/types.hpp

```diff
@@ -40,4 +43,20 @@ using ImuData     = std::array<Float, kNumImus>;
 using EncoderData = std::array<std::uint32_t, kNumEncoders>;
 using KeyenceData = std::array<std::uint32_t, kNumKeyence>;
 
+// data produced by the accelerometer sensor
+struct RawAccelerationData {
+  RawAccelerationData(core::Float x, core::Float y, core::Float z, TimePoint time)
```

Comment:
don't declare the constructor in the header; make a cpp file if that's what it takes

also `const`

then again, i don't think we need a constructor here